### PR TITLE
(CDAP-5311) Improvements to Authorization REST APIs

### PR DESCRIPTION
**Pull request by bdmogal**
- ``/v3/security/authorization/grant`` becomes ``/v3/security/authorization/privileges/grant``
- ``/v3/security/authorization/revoke`` becomes ``/v3/security/authorization/privileges/revoke``
- ``/v3/security/authorization/authorized`` is no longer available

Code cleanup:
- ``AuthorizationClient`` now implements ``Authorizer``
- Removed public from constructors that do not need to be public (are ``@Inject'ed``)
- Made enum fields in Authorization CLI commands case insensitive
- Other general code cleanup around the CLI and proto classes

(CDAP-5326) Added authorization RESTful API paths to ``RouterPathLookup``

Jira: 
[CDAP-5311](https://issues.cask.co/browse/CDAP-5311)
[CDAP-5326](https://issues.cask.co/browse/CDAP-5326)

Build: http://builds.cask.co/browse/CDAP-DUT3784

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java [hunk 1]
```diff
@@ -30,6 +35,17 @@
 
   protected final CLIConfig cliConfig;
 
+  protected static final Function<String, Set<Action>> ACTIONS_STRING_TO_SET = new Function<String, Set<Action>>() {
+    @Override
+    public Set<Action> apply(String input) {
+      ImmutableSet.Builder<Action> resultBuilder = ImmutableSet.builder();
+      for (String action : Splitter.on(",").split(input)) {
```
Comment (gokulavasan) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java]: Will it work if the string is 'read,         write'  ?

File: cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/AuthorizationHandlerTest.java [hunk 2]
```diff
@@ -136,29 +136,72 @@ public Authorizer get() {
...
-      client.revoke(ns1, admin, ImmutableSet.of(Action.READ));
-      Assert.fail();
-    } catch (FeatureDisabledException expected) {
-      Assert.assertEquals("cdap-site.xml", expected.getConfigFile());
-    }
+    final NamespaceId ns1 = Ids.namespace("ns1");
+    final Principal admin = new Principal("admin", Principal.PrincipalType.USER);
+    final Role admins = new Role("admins");
+
+    verifyFeatureDisabled(new DisabledFeatureCaller() {
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/AuthorizationHandlerTest.java]: Can you explain a bit about what these verifyFeatureDisabled calls are doing, and may be a leave a comment too?

**Comment by gokulavasan** [comment]
Reviewed. Couple of comments/questions.

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java (hunk 1 above)
Comment (bdmogal) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java]: Thought about that, but this is the actions string that the user enters in the CLI. Having a space in the middle will fail to parse as the CLI command. So guarding against whitespace is not necessary.

File: cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/AuthorizationHandlerTest.java (hunk 2 above)
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/AuthorizationHandlerTest.java]: So when authorization is disabled, we return 503 with a descriptive message on what setting to enable in which configuration file. This just tests that functionality. I'll add a comment

**Comment by bdmogal** [comment]
@gokulavasan addressed comments. please review again. Thanks!

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java (hunk 1 above)
Comment (rohitsinha54) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java]: Should we not depend on the CLI format and trim on split.

Comment (bdmogal) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java]: ok done.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by bdmogal** [comment]
Thanks @gokulavasan. Squashed commits. Merging.

**Pull request merged**